test/unleash_proxy_client_flutter_test.dart

```diff
@@ -588,6 +588,57 @@ void main() {
     ]);
   });
 
+  test(
+      'set and update context with the same value will not trigger new fetch call',
+      () async {
+    final getMock = GetMock();
+    final unleash = UnleashClient(
+        url: url,
+        clientKey: 'proxy-123',
+        appName: 'flutter-test',
+        sessionIdGenerator: generateSessionId,
+        storageProvider: InMemoryStorageProvider(),
+        fetcher: getMock);
+    await unleash.updateContext(UnleashContext(
+        userId: '123',
+        remoteAddress: 'address',
+        sessionId: 'session',
+        properties: {'customKey': 'customValue'}));
+    // update whole context before start
+    await unleash.updateContext(UnleashContext(
+        userId: '123',
+        remoteAddress: 'address',
+        sessionId: 'session',
+        properties: {'customKey': 'customValue'}));
```

Comment:
Are these two the exact same? Is that the point? the comments seem to indicate that we're doing different things, but my eyes tell me that these look the same. What am I missing here?

Oh, this is just how we set the initial context, so we're checking that setting the new context (which is exactly the same) doesn't cause an update!

Reply:
added a comment explaining this